mne/minimum_norm/tests/test_inverse.py

```diff
@@ -1190,4 +1191,24 @@ def test_inverse_mixed_loose(mixed_fwd_cov_evoked):
     assert_allclose(got_pos, want_pos, atol=1.1e-2)
 
 
+@testing.requires_testing_data
+def test_sss_rank():
+    """Test passing rank explicitly during inverse computation."""
+    # make raw match the fwd and cov, doesn't matter that they are mismatched
+    raw = mne.io.read_raw_fif(fname_sss).pick_types(meg=True)
+    raw.rename_channels(
+        {ch_name: f'{ch_name[:3]} {ch_name[3:]}' for ch_name in raw.ch_names})
+    fwd = mne.read_forward_solution(fname_fwd)
+    cov = mne.read_cov(fname_cov)
+    with pytest.warns(RuntimeWarning, match='rank as it exceeds.*302 > 67'):
+        inv = make_inverse_operator(raw.info, fwd, cov)
+    rank = (inv['noise_cov']['eig'] > 0).sum()
+    assert rank == 302
+    # should not warn
+    inv = make_inverse_operator(raw.info, fwd, cov, rank=dict(meg=67))
```

Comment:
on `master` this warns about the data-driven rank calc 302 > 67 (which it shouldn't -- rank is passed explicitly)